### PR TITLE
NozomiNetworks: first commit

### PR DESCRIPTION
**Pull request by averbn**

### CODE REVIEW

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                    },
+                    "type": "CopyableLabel"
+                }
+            ]
+        },
+        {
+            "title": "4. Secure your machine ",
+            "description": "Make sure to configure the machine's security according to your organization's security policy\n\n\n[Learn more >](https://aka.ms/SecureCEF)"
+        }
+    ]
```
Comment (devikamehra) [inline Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:120]: "metadata" parameter is missing from this template. Please refer to CEF Data connector template to gain more insights. (https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_CEF_template.json)

Comment (devikamehra) [inline Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:120]: "metadata" parameter is missing from this template. Please refer to CEF Data connector template to gain more insights. (https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_CEF_template.json)

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
+{
+    "id": "NozomiNetworks",
+    "title": "Nozomi Networks",
+    "publisher": "Nozomi Networks",
+    "descriptionMarkdown": "The [Nozomi Networks](https://www.nozominetworks.com/) data connector provides the capability to ingest Nozomi Networks Events into Azure Sentinel. Refer to the Nozomi Networks PDF documentation for more information.",
```
Comment (devikamehra) [inline Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:5]: Include the link to Nozomi PDF Documentation here.

Comment (devikamehra) [inline Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:5]: Include the link to Nozomi PDF Documentation here.

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
+{
```
Comment (devikamehra) [inline Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:1]: Follow the guideline to Name the data connector file. 
https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Data%20Connectors%20Template%20Guidance.md

Comment (devikamehra) [inline Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:1]: Follow the guideline to Name the data connector file. 
https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Data%20Connectors%20Template%20Guidance.md

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    "id": "NozomiNetworks",
+    "title": "Nozomi Networks",
```
Comment (devikamehra) [inline Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:3]: Title should be in the format "PROVIDER NAME APPLIANCE NAME". Appliance Name is missing.

Comment (devikamehra) [inline Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:3]: Title should be in the format "PROVIDER NAME APPLIANCE NAME". Appliance Name is missing.

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    ],
+    "sampleQueries": [
+        {
+            "description" : "Top 10 Devices", 
+            "query": "NozomiNetworksEvents\n | summarize count() by DvcHostname\n | top 10 by count_"
+        }
+    ],
+    "dataTypes": [
+        {
+            "name": "CommonSecurityLog (Nozomi Networks)",
```
Comment (devikamehra) [inline Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:22]: Remove the space Between Nozomi and Networks.

Comment (devikamehra) [inline Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:22]: Remove the space Between Nozomi and Networks.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: .script/tests/KqlvalidationsTests/CustomTables/NozomiNetworksEvents [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+          "Name":"MitreAttackTechniques",
+          "Type":"String"
+       },
+       {
+          "Name":"Risk",
+          "Type":"String"
+       },
+       {
+          "Name":"IsSecurity",
+          "Type":"String"
```
[the remark this answers is not in the document]
Comment (averbn) [reply .script/tests/KqlvalidationsTests/CustomTables/NozomiNetworksEvents:74]: done

[the remark this answers is not in the document]
Comment (averbn) [reply .script/tests/KqlvalidationsTests/CustomTables/NozomiNetworksEvents:74]: done

File: .script/tests/KqlvalidationsTests/CustomTables/NozomiNetworksEvents [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+          "Name":"SrcMacAddr",
+          "Type":"String"
+       },
+       {
+          "Name":"DstPortNumber",
+          "Type":"String"
+       },
+       {
+          "Name":"SrcPortNumber",
+          "Type":"String"
```
[the remark this answers is not in the document]
Comment (averbn) [reply .script/tests/KqlvalidationsTests/CustomTables/NozomiNetworksEvents:106]: done

[the remark this answers is not in the document]
Comment (averbn) [reply .script/tests/KqlvalidationsTests/CustomTables/NozomiNetworksEvents:106]: done

File: Sample Data/NozomiNetworksEvents.json [hunk 8]
```diff
@@ -0,0 +1,118 @@
+[
```
[the remark this answers is not in the document]
Comment (averbn) [reply Sample Data/NozomiNetworksEvents.json:1]: done

[the remark this answers is not in the document]
Comment (averbn) [reply Sample Data/NozomiNetworksEvents.json:1]: done

File: Solutions/NozomiNetworks/Parsers/NozomiNetworksEvents.txt [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+| extend EventProductVersion = DeviceVersion
+| extend EventSeverity = LogSeverity
+| extend SrcIpAddr = SourceIP
+| extend SrcHostName = SourceHostName
+| extend DstIpAddr = DestinationIP
+| extend DstHostname = DestinationHostName
+| extend SrcUserName = SourceUserName
+| extend EventMessage = Message
+| extend EventSubType = DeviceEventClassID
+| extend Startmsec = extract(@'start=(\w+).*', 1, AdditionalExtensions)
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/NozomiNetworks/Parsers/NozomiNetworksEvents.txt:18]: done

File: Solutions/NozomiNetworks/Parsers/NozomiNetworksEvents.txt [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+| extend IsSecurity = iff(DeviceCustomString2Label == "isSecurity", DeviceCustomString2, '')
+| extend EventOriginalUid = DeviceCustomString3
+| extend TriggerType = DeviceCustomString4
+| extend Parents = DeviceCustomString5
+| extend NetworkApplicationProtocol = ApplicationProtocol
+| extend DstMacAddr = DestinationMACAddress 
+| extend SrcMacAddr = SourceMACAddress
+| extend DstPortNumber= DestinationPort
+| extend SrcPortNumber= SourcePort
+| extend EventType = iff(DeviceCustomString1Label contains "Risk", "Alert", iff(DeviceCustomString1Label contains "Browser","Audit Log","Health Log"))
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/NozomiNetworks/Parsers/NozomiNetworksEvents.txt:34]: done

File: Solutions/NozomiNetworks/Parsers/NozomiNetworksEvents.txt [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+| project   TimeGenerated,
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/NozomiNetworks/Parsers/NozomiNetworksEvents.txt:35]: done

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json (hunk 1 above)
Comment (averbn) [reply Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:120]: done

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json (hunk 2 above)
Comment (averbn) [reply Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:5]: No direct link to the pdf doc. User can find them after login to the account

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json (hunk 3 above)
Comment (averbn) [reply Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:1]: done

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json (hunk 4 above)
Comment (averbn) [reply Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:3]: done

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json (hunk 5 above)
Comment (averbn) [reply Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json:22]: done

**Comment by v-spadarthi** [comment]
@averbn : We can see the some of the comments are not addressed please resolve those comments.

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json (hunk 4 above)
Comment (averbn) [reply Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json]: done

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json (hunk 1 above)
Comment (averbn) [reply Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json]: done

**Comment by averbn** [comment]
>

**Comment by averbn** [comment]
> @averbn : We can see the some of the comments are not addressed please resolve those comments.

@v-spadarthi  It seems all comments are done. Сan you point out what is wrong?

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json (hunk 5 above)
Comment (averbn) [reply Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json]: done

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json (hunk 1 above)
Comment (averbn) [reply Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json]: image
For our connector it should be:
image
Is it ok?

File: Sample Data/NozomiNetworksEvents.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (averbn) [reply Sample Data/NozomiNetworksEvents.json:1]: Log sample in the zip archive:
image

File: Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json (hunk 1 above)
Comment (averbn) [reply Solutions/NozomiNetworks/Data Connectors/Connector_CEF_NozomiNetworks.json]: metadata section for this connector:
image

**Comment by devikamehra** [comment]
> @v-spadarthi It seems all comments are done. Сan you point out what is wrong?

Name under source object is missing in metadata (Highlighted below).

"metadata": {
        "id": "",
        "version": "",
        "kind": "dataConnector",
        "source": {
            "kind": "",
            **"name": "Name of the content source. The repo name, solution name, LA workspace name etc."**
        },
        "author": {
            "name": ""
        },
        "support": {
            "tier": "",
            "name": ""
        }
    }

**Comment by averbn** [comment]
> > @v-spadarthi It seems all comments are done. Сan you point out what is wrong?
> 
> Name under source object is missing in metadata (Highlighted below).
> 
> "metadata": { "id": "", "version": "", "kind": "dataConnector", "source": { "kind": "", **"name": "Name of the content source. The repo name, solution name, LA workspace name etc."** }, "author": { "name": "" }, "support": { "tier": "", "name": "" } }

Thanks! The solution name was added.

**Review by devikamehra — APPROVED** [review]
All required changes are in place.

File: Sample Data/NozomiNetworksEvents.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (averbn) [reply Sample Data/NozomiNetworksEvents.json:1]: https://github.com/Azure/Azure-Sentinel/raw/b2d467055911664826132c873b5e1bb2faa18179/Sample%20Data/NozomiNetworksCEF.zip

**Comment by averbn** [comment]
> Hey @averbn, can you please remove zip file from sample data folder, and also please move Sample Data/NozomiNetworksCEF.zip into [Azure-Sentinel](https://github.com/Azure/Azure-Sentinel)/[Sample Data](https://github.com/Azure/Azure-Sentinel/tree/master/Sample%20Data)/CEF/

Sure, done

**Comment by v-sabiraj** [comment]
Hey @averbn, the zip file is still there and also sample data is not moved to CEF folder, will you please check and update, thanks.
image

**Review by v-sabiraj — APPROVED** [review]

**Pull request merged**